tests/Test/Xrefcheck/IgnoreAnnotationsSpec.hs

```diff
@@ -7,17 +7,41 @@ module Test.Xrefcheck.IgnoreAnnotationsSpec where
 
 import Universum
 
-import Test.Hspec (Spec, describe, it, shouldBe)
+import CMarkGFM (PosInfo (..))
+import Test.Hspec (Spec, describe, it, shouldBe, shouldReturn)
 
 import Test.Xrefcheck.Util
 import Xrefcheck.Core
+import Xrefcheck.Scan
+import Xrefcheck.Scanners.Markdown
 
 spec :: Spec
 spec = do
   describe "Parsing failures" $ do
-    it "Check if parsing incorrect markdowns produce exceptions" $ do
-      areIncorrect <- mapM (isIncorrectMD GitHub) failPaths
-      or areIncorrect `shouldBe` True
+    it "Check if broken link annotation produce error" do
+      let file = "tests/markdowns/with-annotations/no_link.md"
+      getErrs file `shouldReturn`
+        makeError (Just $ PosInfo 7 1 7 31) file Link ""
+    it "Check if broken paragraph annotation produce error" do
+      let file = "tests/markdowns/with-annotations/no_paragraph.md"
+      getErrs file `shouldReturn`
+        makeError (Just $ PosInfo 7 1 7 35) file Paragraph "HEADING"
+    it "Check if broken ignore file annotation produce error" do
+      let file = "tests/markdowns/with-annotations/unexpected_ignore_file.md"
+      getErrs file `shouldReturn`
+        makeError (Just $ PosInfo 9 1 9 30) file File ""
+    it "Check if broken unrecognised annotation produce error" do
+      let file = "tests/markdowns/with-annotations/unrecognised_option.md"
+      getErrs file `shouldReturn`
+        makeError (Just $ PosInfo 7 1 7 46) file None "unrecognised-option"
+    it "Check if ignore link annotation produce error when placed at the end of file" do
+      let file = "tests/markdowns/with-annotations/no_link_eof.md"
+      getErrs file `shouldReturn`
+        makeError (Just $ PosInfo 9 1 9 31) file Link ""
+    it "Check if ignore paragraph annotation produce error when placed at the end of file" do
+      let file = "tests/markdowns/with-annotations/no_paragraph_eof.md"
+      getErrs file `shouldReturn`
+        makeError (Just $ PosInfo 9 1 9 36) file Paragraph "EOF"
```

Comment:
I think it would be even better if these 2 new tests were written in bats instead, because then we can see whether the error message printed to stdout is sensible.

Reply:
Tried this manually, it seems there's a double space in `"but found^^EOF"`.